doc/user/sql-grammar/sql-grammar.bnf

```diff
@@ -8,6 +8,9 @@ alter_index ::=
     )
     | 'RESET' '(' field ( ',' field )* ')'
   )
+array_agg ::=
+  'array_agg' '(' value ('ORDER BY' colname)?
```

Comment:
All of these order by things are a bit off. They should be identical to the other ORDER BY in this file.

```
array_agg ::=
    'array_agg' '(' value
    ( 'ORDER' 'BY' col_ref ( 'ASC' | 'DESC' )? ( ',' col_ref ( 'ASC' | 'DESC' )? )* )?
    ')'
```